src/bindings/lib/bindings.c

```diff
@@ -40,13 +40,7 @@ typedef struct _tree {
 } tree_W;
 
 static void finalize_tree(value v) {
-  tree_W *p;
-  p = (tree_W *)Data_custom_val(v);
-  //TODO: ts_tree_delete(p->tree);
-  // this caused some segfaults, probably during Gc after
-  // analyzing many Ruby files. We go around this bug by
-  // running the Ruby parser in a separate process so segfaults
-  // or here memory leak do not reach the main semgrep-core process.
+  // TODO I assume we can then delete this function? But pushing to github first
```

Comment:
You need to keep the original code commented as well as the big comment and add after an explanation that people should
use the delete function.